dds/src/implementation/rtps/types.rs

```diff
@@ -279,38 +300,51 @@ pub const PROTOCOLVERSION_2_3: ProtocolVersion = ProtocolVersion { major: 2, min
 pub const PROTOCOLVERSION_2_4: ProtocolVersion = ProtocolVersion { major: 2, minor: 4 };
 
 impl ProtocolVersion {
-    pub fn new(major: u8, minor: u8) -> Self {
+    pub const fn new(major: u8, minor: u8) -> Self {
         Self { major, minor }
     }
-    pub fn major(&self) -> u8 {
+    pub const fn major(&self) -> u8 {

```

Comment:
What is the rationale behind making the getters const? Are they used in const contexts?

Reply:
Not sure if they are used in const contexts. But the compiler can make them const at compile time already. But maybe to restrictive for future implementations?

Reply:
I was indeed thinking of restrictions but now that I think again these types are not exposed to the user so we can do whatever we want